File: 2724-convert-an-array-into-a-2d-array-with-conditions/2724-convert-an-array-into-a-2d-array-with-conditions.cpp
#pragma GCC optimize("O3")

class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        int map[201] = {};
        vector<vector<int>> res;
        for (const auto i: nums) {
            ++map[i];
            if (map[i] > res.size()) {
                res.push_back({i});
            } else {
                res[map[i] - 1].push_back(i);
            }
        }

        return res;
    }
};

File: 2724-convert-an-array-into-a-2d-array-with-conditions/2724-convert-an-array-into-a-2d-array-with-conditions.go
func findMatrix(nums []int) [][]int {
    m := make([]int, 201);
    res := make([][]int, 1);

    for _, n := range nums {
        m[n] += 1;
        if m[n] > len(res) {
            res = append(res, []int{n});
        } else {
            res[m[n] - 1] = append(res[m[n] - 1], n);
        }
    }

    return res;
}